utils/src/main/java/com/cloud/utils/nio/Link.java

```diff
@@ -453,115 +449,150 @@ public static SSLContext initSSLContext(boolean isClient) throws GeneralSecurity
         return sslContext;
     }
 
-    public static void doHandshake(SocketChannel ch, SSLEngine sslEngine, boolean isClient) throws IOException {
-        if (s_logger.isTraceEnabled()) {
-            s_logger.trace("SSL: begin Handshake, isClient: " + isClient);
+    public static ByteBuffer enlargeBuffer(ByteBuffer buffer, final int sessionProposedCapacity) {
```

Comment:
Add two ``Preconditions.checkArgument`` checks to verify that the passed ``buffer`` is not ``null`` and that ``sessionProposedCapacity`` is greater than zero.